core/src/WsTrust/MexParser.cs

```diff
@@ -79,61 +77,38 @@ public static async Task<WsTrustAddress> FetchWsTrustAddressFromMexAsync(string
 
         internal static async Task<XDocument> FetchMexAsync(string federationMetadataUrl, RequestContext requestContext)
         {
-            XDocument mexDocument;
-            try
-            {
-                IHttpClient request = new HttpClientWrapper(federationMetadataUrl, requestContext);
-                using (var response = await request.GetResponseAsync().ConfigureAwait(false))
-                {
-                    mexDocument = XDocument.Load(EncodingHelper.GenerateStreamFromString(response.ResponseString), LoadOptions.None);
-                }
-            }
-            catch (HttpRequestWrapperException ex)
-            {
-                throw new AdalServiceException(AdalError.AccessingWsMetadataExchangeFailed, ex);
+            var uri = new UriBuilder(federationMetadataUrl);
+            var httpResponse = await HttpRequest.SendGet(uri.Uri, null, requestContext).ConfigureAwait(false);
+            if (httpResponse.StatusCode != System.Net.HttpStatusCode.OK) {
+                throw CoreExceptionFactory.Instance.GetClientException(
+                    CoreErrorCodes.AccessingWsMetadataExchangeFailed,
+                    string.Format(CultureInfo.CurrentCulture,
+                        CoreErrorMessages.HttpRequestUnsuccessful,
+                        (int)httpResponse.StatusCode, httpResponse.StatusCode));
             }
-            catch (XmlException ex)
-            {
-                throw new AdalException(AdalError.ParsingWsMetadataExchangeFailed, ex);
-            }
-
-            return mexDocument;
+            return XDocument.Parse(httpResponse.Body, LoadOptions.None);
         }
 
         internal static WsTrustAddress ExtractWsTrustAddressFromMex(XDocument mexDocument, UserAuthType userAuthType, RequestContext requestContext)
+        // This method would return null if the wanted policy is not found in mexDocument. It could potentially throw XmlException.
```

Comment:
>// This method would return [](start = 8, length = 27)

nit: imo, this comment should be on the method and not inside the method.

Reply:
I was not explaining myself clearly. On the method: meaning an XML comment on the method. This comment is useless for the caller for who this seems to be an interesting comment on the expectations of the code. If I'm already reading the code I can probably figure it out. 
Please move it to the return or remarks section of the xml comment if it is still relevant.